evm/src/modules/IntentGatewayV2.sol

```diff
@@ -0,0 +1,899 @@
+    address dispatcher;
+    /// @dev Protocol fee in basis points (BPS) deducted from filler-provided tokens
+    uint256 protocolFeeBps;
+}
+
+/**
+ * @dev Struct representing the body of a request.
+ */
+struct RequestBody {
+    /// @dev Represents the commitment of an order. This is typically a hash that uniquely identifies the order.
+    bytes32 commitment;
+    /// @dev Stores the identifier for the beneficiary.
+    bytes32 beneficiary;
+    /// @dev An array of token identifiers. Each element in the array represents a unique token involved in the order.
+    TokenInfo[] tokens;
+}
+
+/**
+ * @notice A struct representing the options for filling an intent.
+ * @dev This struct is used to specify various parameters and options
+ *      when filling an intent in the IntentGateway contract.
+ */
+struct FillOptions {
+    /// @dev The fee paid to the relayer for processing transactions.
+    uint256 relayerFee;
+}
+
+/**
+ * @dev Struct representing the options for canceling an intent.
+ */
+struct CancelOptions {
+    /// @dev The fee paid to the relayer for processing transactions.
+    uint256 relayerFee;
+    /// @dev Stores the height value.
+    uint256 height;
+}
+
+/**
+ * @dev Request from hyperbridge for adding a new deployment of IntentGateway
+ */
+struct NewDeployment {
+    /// @dev Identifier for the state machine.
+    bytes stateMachineId;
+    /// @dev A bytes32 variable to store the gateway identifier.
+    bytes32 gateway;
+}
+
+/**
+ * @title IntentGatewayV2
+ * @author Polytope Labs (hello@polytope.technology)
+ *
+ * @dev The IntentGateway allows for the creation and fulfillment of cross-chain orders.
+ */
+contract IntentGatewayV2 is HyperApp {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev Enum representing the different kinds of incoming requests that can be executed.
+     */
+    enum RequestKind {
+        /// @dev Identifies a request for redeeming an escrow.
+        RedeemEscrow,
+        /// @dev Identifies a request for recording new contract deployments
+        NewDeployment,
+        /// @dev Identifies a request for updating parameters.
+        UpdateParams,
+        /// @dev Identifies a request for redeeming an escrow.
+        RefundEscrow,
+        /// @dev Identifies a request for collecting fees.
+        CollectFees
+    }
+
+    /**
+     * @dev Address constant for transaction fees, derived from the keccak256 hash of the string "txFees".
+     * This address is used to store or reference the transaction fees within the contract.
+     */
+    address private constant TRANSACTION_FEES = address(uint160(uint256(keccak256("txFees"))));
+
+    /**
+     * @notice Constant representing a filled slot in big endian format
+     * @dev Hex value 0x05 padded with leading zeros to fill 32 bytes
+     */
+    bytes32 constant FILLED_SLOT_BIG_ENDIAN_BYTES =
+        hex"0000000000000000000000000000000000000000000000000000000000000005";
```

Comment:
I think we might have to update this filled slot @royvardhan

Reply:
Forge can list it `forge inspect IntentGatewayV2 storage`